lectures/sympy.md

```diff
@@ -0,0 +1,924 @@
+
+```{code-cell} ipython3
+X = Exponential(x, λ)
+```
+
+```{code-cell} ipython3
+moment(X, 1)
+```
+
+```{code-cell} ipython3
+E(X**t)
+```
+
++++ {"user_expressions": []}
+
+Using `integrate` function, we can derive the cumulative density function of the exponential distribution with $\lambda = 0.5$
+
+```{code-cell} ipython3
+lamda_pdf = pdf.subs(λ, 1/2)
+lamda_pdf
+```
+
+```{code-cell} ipython3
+integrate(lamda_pdf, (x, 0, 4))
+```
+
++++ {"user_expressions": []}
+
+## Plotting
+
+SymPy provides a powerful plotting feature. 
+
+First we plot a simple function using the `plot` function
+
+```{code-cell} ipython3
+f = sin(2 * sin(2 * sin(2 * sin(x))))
+p = plot(f, (x, -10, 10), show=False)
+p.title = 'A Simple Plot'
+p.show()
+```
+
++++ {"user_expressions": []}
+
+Similar to Matplotlib, SymPy provides an interface to customize the graph
+
+```{code-cell} ipython3
+plot_f = plot(f, (x, -10, 10), 
+              xlabel='', ylabel='', legend = True, show = False)
+plot_f[0].label = 'f(x)'
+df = diff(f)
+plot_df = plot(df, (x, -10, 10), legend = True, show = False)
+plot_df[0].label = 'f\'(x)'
+plot_f.append(plot_df[0])
+plot_f.show()
+```
+
++++ {"user_expressions": []}
+
+It also supports plotting implicit functions and visualizing inequalities
+
+```{code-cell} ipython3
+p = plot_implicit(Eq((1/x + 1/y)**2, 1))
+```
+
+```{code-cell} ipython3
+p = plot_implicit(And(2*x + 5*y <= 30, 4*x + 2*y >= 20), (x, -1, 10), (y, -10, 10))
+```
+
++++ {"user_expressions": []}
+
+It also supports visualizations in three-dimensional space
+
+```{code-cell} ipython3
+p = plot3d(cos(2*x + y))
+```
+
++++ {"user_expressions": []}
+
+## Applications
+
+In this section, we apply SymPy to an economic model that explores the wage gap between college and high school graduates. 
+
+### Defining the Symbols
+
+The first step in symbolic computation is to define the symbols that represent the variables in our equations. In our model, we have the following variables:
+
+ * $R$ The gross rate of return on a one-period bond
+
+ * $t = 0, 1, 2, \ldots T$ Denote the years that a person either works or attends college
+ 
+ * $T$ denotes the last period that a person  works
+ 
+ * $w$ The wage of a high school graduate
+ 
+ * $g$ The growth rate of wages
+
+ * $D$ The upfront monetary costs of going to college
+ 
+ * $\phi$ The wage gap, defined as the ratio of the wage of a college graduate to the wage of a high school graduate
+ 
+Let's define these symbols using sympy
+
+```{code-cell} ipython3
+# Define the symbols
+w, R, g, D, phi = symbols('w R g D phi')
+```
+
++++ {"user_expressions": []}
+
+### Defining the Present Value Equations
+We calculate the present value of the earnings of a high school graduate and a college graduate. For a high school graduate, we sum the discounted wages from year 1 to $T = 4$. For a college graduate, we sum the discounted wages from year 5 to $T = 4$, considering the cost of college $D$ paid upfront
```

Comment:
It can be useful to add a plot that shows the various flows `+/-` along a timeline. Would that be possible?